aptos-move/block-executor/src/errors.rs

```diff
@@ -13,28 +14,29 @@ pub enum IntentionalFallbackToSequential {
     /// TODO: (short-mid term) relax the limitation, and (mid-long term) provide proper multi-versioning
     /// for code (like data) for the cache.
     ModulePathReadWrite,
-    /// We defensively check certain resource group related invariant violations.
-    ResourceGroupError(String),
+    /// We defensively check resource group serialization error in the commit phase.
+    /// TODO: should trigger invariant violation in the transaction itself.
+    ResourceGroupSerializationError(String),
```

Comment:
When there is a serialization error for resource groups, I understand we need to fallback to sequential execution. Sequential execution may result in a different resource group that doesn't have the serialization problem. Shouldn't we fallback to sequential execution when there is a serialization error for resources as well?

Reply:
Hm, @vusirikala, do you mean 
1) I have a resource with delayed fields
2) I process it so it can be serialized
3) during patching serialization fails?

Reply:
We talked about it, confusing part was that resource group serialization happens in commit now, resources during session, and fallback is to skip, not to try to serialize better (unless it does happen to be concurrency-related, which we are handling that way anyway - and we will still get the error/alert).